tests/unit/ml/test_golden_sql.py

```diff
@@ -209,3 +210,55 @@ def test_logistic_regression_score(mock_session, bqml_model, mock_X, mock_y):
     mock_session.read_gbq.assert_called_once_with(
         "SELECT * FROM ML.EVALUATE(MODEL `model_project`.`model_dataset`.`model_id`,\n  (input_X_y_sql))"
     )
+
+
+def test_decomposition_mf_default_fit(bqml_model_factory, mock_session, mock_X):
+    model = decomposition.MatrixFactorization(
+        num_factors=34,
+        feedback_type="explicit",
+        user_col="user_id",
+        item_col="item_col",
+        rating_col="rating_col",
+        l2_reg=9.83,
+    )
+    model._bqml_model_factory = bqml_model_factory
+    model.fit(mock_X)
+
+    mock_session._start_query_ml_ddl.assert_called_once_with(
+        "CREATE OR REPLACE MODEL `test-project`.`_anon123`.`temp_model_id`\nOPTIONS(\n  model_type='matrix_factorization',\n  feedback_type='explicit',\n  user_col='user_id',\n  item_col='item_col',\n  rating_col='rating_col',\n  l2_reg=9.83,\n  num_factors=34)\nAS input_X_no_index_sql"
+    )
+
+
+def test_decomposition_mf_predict(mock_session, bqml_model, mock_X):
+    model = decomposition.MatrixFactorization(
+        num_factors=34,
+        feedback_type="explicit",
+        user_col="user_id",
+        item_col="item_col",
+        rating_col="rating_col",
+        l2_reg=9.83,
+    )
+    model._bqml_model = bqml_model
+    model.predict(mock_X)
+
+    mock_session.read_gbq.assert_called_once_with(
+        "SELECT * FROM ML.RECOMMEND(MODEL `model_project`.`model_dataset`.`model_id`,\n  (input_X_sql))",
+        index_col=["index_column_id"],
```

Comment:
Why is `index_col` here? I only see `trial_id` in https://cloud.google.com/bigquery/docs/reference/standard-sql/bigqueryml-syntax-recommend?hl=en